ui/e2e/setup.js

```diff
@@ -0,0 +1,28 @@
+// setup.js
+const { chromium } = require("playwright");
+const { extractExtensionPackage } = require("./utils");
+const { MM_CONFIG, KEPLR_CONFIG } = require("./config.js");
+const path = require("path");
+const fs = require("fs");
+
+beforeAll(async () => {
+  await extractExtensionPackage(MM_CONFIG.id);
+  await extractExtensionPackage(KEPLR_CONFIG.id);
+  const pathToKeplrExtension = path.join(__dirname, KEPLR_CONFIG.path);
+  const pathToMmExtension = path.join(__dirname, MM_CONFIG.path);
+  const userDataDir = path.join(__dirname, "./playwright");
+  // need to rm userDataDir or else will store extension state
+  if (fs.existsSync(userDataDir)) {
+    fs.rmdirSync(userDataDir, { recursive: true });
+  }
+
+  context = await chromium.launchPersistentContext(userDataDir, {
+    // headless required with extensions. xvfb used for ci/cd
+    headless: false,
+    args: [
+      `--disable-extensions-except=${pathToKeplrExtension},${pathToMmExtension}`,
+      `--load-extension=${pathToKeplrExtension},${pathToMmExtension}`,
+    ],
+  });
+  [page] = await context.pages();
```

Comment:
This is a little thing but I hadn't read your description of this PR and was looking at the code and it was only once I came across this that I understood that you were saving page globally. It is not the most common js syntax to de structure an array to a global var but it is fine here. I think I would just love either a comment as I scan the code to ensure it is not a mistake. Something like `// exposing "page" object globally` and `// exposing "context" globally` alternatively you could do the following which also signals intent:
```
[globalThis.page] = await context.pages();
```

Reply:
Sure, thanks for this comment. For now I put the comments as you requested and in the near future, we would utilize global objects with `this.global.page` and `this.global.context` as we introduce custom environments like described in https://github.com/playwright-community/jest-playwright#usage-with-custom-testenvironment